opentelemetry-sdk/src/logs/logger_provider.rs

```diff
@@ -346,9 +338,43 @@ mod tests {
             Ok(())
         }
     }
+
+    #[derive(Debug, Clone)]
+    struct TestProcessorForResource {
+        resource: Arc<Mutex<Resource>>,
```

Comment:
Could we get rid of the `Arc<Mutex<T>>` wrapper now that you have updated the method to require a `&mut self`?